discrete_math/representations_of_numbers/representation.cpp

```diff
@@ -0,0 +1,121 @@
+#include "representation.h"
+
+using namespace representation;
+
+int Rep::load_numerator() {
+	int x;
+
+	std::cout << "\nPodaj licznik ulamka:\t";
+	std::cin >> x;
+
+	if (std::cin.fail()) {
+		std::cout << "\nPodana wartosc jest bledna.";
+		std::cin.clear();
+		std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+		return 0;
+	}
+
+	return x;
+}
+
+int Rep::load_denominator() {
+	int x;
+
+	std::cout << "\nPodaj mianownik ulamka:\t";
+	std::cin >> x;
+
+	if (std::cin.fail() || x == 0) {
+		std::cout << "\nPodana wartosc jest bledna.";
+		std::cin.clear();
+		std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+		return 1;
+	}
+
+	return x;
+}
+
+int Rep::load_base() {
+	int x;
+
+	std::cout << "\nPodaj baze rozwiniecia ulamka:\t";
+	std::cin >> x;
+
+	if (std::cin.fail() || x < 2) {
+		std::cout << "\nPodana wartosc jest bledna.";
+		std::cin.clear();
+		std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+		
+		return 10;
+	}
+
+	return x;
+}
+
+Fraction Rep::create_fraction(int x, int y) {
+	Fraction fraction{ x, y};
+
+	return fraction;
+}
+
+Expansion Rep::create_expansion(int x) {
+	Expansion expansion{ x };
+
+	return expansion;
+}
+
+std::string Rep::expansion_in_base(Fraction f, Expansion e) {

```

Comment:
too long definition of function - split code below into many functions - each function cannot contain more than 8 lines of code